src/runtime_src/core/tools/common/XBUtilities.cpp

```diff
@@ -284,7 +284,8 @@ XBUtilities::wrap_paragraphs( const std::string & _unformattedString,
 void
 XBUtilities::parse_device_indices(std::vector<uint16_t> &device_indices, const std::string &device)
 {
-  if(boost::iequals(device, "all")) {
+  //if no device is passed or "all" is specified, parse all devices
+  if(boost::iequals(device, "all") || device.empty()) {

```

Comment:
(minor item)
This functionality does work.  You might wish to consider to break this logic into to separate lines for future EoU reading.  For example:
```
bool parse_all_devices = device.empty() | boost::iequals("device", "all");
if (parse_all_devices) {
```